### ISSUE TITLE
tgn example can only use cpu

### ISSUE THREAD

**Issue by dwang55** [dmlc/dgl#2970]
## 🚀 Feature

Support for using gpu to train tgn model

## Motivation

There is an example of tgn here: https://github.com/dmlc/dgl/tree/master/examples/pytorch/tgn. But it can only use cpu to train the model. It's much slower than other implementation(more than 10x times). I am new to dgl so it's a little hard for me to modify the code.
It seems to be related about EdgeCollator which leads it difficult to implement a GPU version.

**Comment by BarclayII** [comment]
You are correct.  By the time TGN is contributed DGL cannot do `compact_graphs` on GPU, nor perform reservoir sampling as in other implementations.  We are aware of this problem, and it is on our backlog.
